src/openforms/js/components/admin/form_design/form-creation-form.js

```diff
@@ -156,29 +158,32 @@ function reducer(draft, action) {
         case 'FIELD_CHANGED': {
             const { name, value } = action.payload;
             // names are prefixed like `form.foo` and `literals.bar`
-            const [prefix, fieldName] = name.split('.');
+            const [prefix, ...rest] = name.split('.');
+            const fieldName = rest.join('.');
 
             switch (prefix) {
                 case 'form': {
-                    draft.form[fieldName] = value;
-                    // remove any validation errors
-                    draft.validationErrors = draft.validationErrors.filter(([key]) => key !== name);
-                    if (!draft.validationErrors.length && draft.tabsWithErrors.includes('form')) {
-                        draft.tabsWithErrors = draft.tabsWithErrors.filter(tab => tab !== 'form');
-                    }
+                    set(draft.form, fieldName, value);
                     break;
                 }
                 case 'literals': {
                     draft.literals[fieldName].value = value;
                     break;
                 }
-                case 'submissionsRemovalOptions':
-                    draft.form.submissionsRemovalOptions[fieldName] = value;
-                    break;
                 default: {
                     throw new Error(`Unknown prefix: ${prefix}`);
                 }
             }
+
+            // remove any validation errors
+            draft.validationErrors = draft.validationErrors.filter(([key]) => key !== name);
+
+            // check which tabs still need the marker and which don't
+            const errorsPerTab = groupBy(draft.validationErrors, ([key]) => {
+                const [prefix, fieldPrefix] = key.split('.');
+                return FORM_FIELDS_TO_TAB_NAMES[fieldPrefix];
+            });
+            draft.tabsWithErrors = draft.tabsWithErrors.filter(tabId => tabId in errorsPerTab);
```

Comment:
This seems to work for the first tab, but not for the registratie tab: I enter an invalid email, submit, update the invalid field, but the error seems to keep showing.

Reply:
time to fix that :-) I tested with the form meta & data cleanup config tabs, so might have missed something :+1: